### PR TITLE
TST: clean old sql tests

### PR DESCRIPTION
**Pull request by fangchenli**
remove skipped mysql tests, simplify old sqlite tests

a small step towards #40686

### CODE REVIEW

File: pandas/tests/io/test_sql.py [hunk 1]
```diff
@@ -2819,301 +2795,7 @@ def clean_up(test_table_to_drop):
...
-            (4, "D"),
-            (5, "E"),
-        ]
-        clean_up(table_name)
-
-
-@pytest.mark.single
-@pytest.mark.db
-@pytest.mark.skip(
-    reason="gh-13611: there is no support for MySQL if SQLAlchemy is not installed"
```
Comment (mroeschke) [inline pandas/tests/io/test_sql.py:2853]: Just curious, is is not possible for us to test MySQL with these test or are these covered in different tests?

Comment (fangchenli) [reply pandas/tests/io/test_sql.py:2853]: These are covered in different tests using sqlalchemy. My guess is many years ago pandas's SQL functions were implemented directly on top of SQL drivers like pymysql. And after refactoring using sqlalchemy, those old tests were skipped but not removed for some reason.

**Review by mroeschke — APPROVED** [review]

File: pandas/tests/io/test_sql.py [hunk 2]
```diff
@@ -2616,31 +2611,44 @@ def format_query(sql, *args):
...
+    return None if res is None else list(res)
 
 
-@pytest.mark.single
-class TestXSQLite(SQLiteMixIn):
-    @pytest.fixture(autouse=True)
-    def setup_method(self, request, datapath):
-        self.method = request.function
+class TestXSQLite:
+    def setup_method(self):
```
[the remark this answers is not in the document]
Comment (fangchenli) [reply pandas/tests/io/test_sql.py:2618]: Yes, I'm going to fixturize all sql tests. will do it in small steps.

[the remark this answers is not in the document]
Comment (jreback) [reply pandas/tests/io/test_sql.py:2618]: kk great

**Pull request merged**